wremnants/datasets/datagroupsLowPU.py

```diff
@@ -1,342 +1,270 @@
 from utilities import boostHistHelpers as hh, logging
 from wremnants import histselections as sel
 from wremnants.datasets import datasetsLowPU
-from wremnants.datasets.datagroups import datagroups
-import lz4.frame
-import pickle
-import narf
-import ROOT
+from wremnants.datasets.datagroups import Datagroups
 import hist
 
 logger = logging.child_logger(__name__)
 
-class datagroupsLowPU(datagroups):
-    isW = False
-    def __init__(self, infile, combine=False, flavor="", excludeProcGroup=None, filterProcGroup=None):
-        self.datasets = {x.name : x for x in datasetsLowPU.getDatasets(flavor=flavor, filt=filterProcGroup, excl=excludeProcGroup)}
-        super().__init__(infile, combine)
-        #self.lumi = 0.199269742
-        self.hists = {} # container storing temporary histograms
+def fakeHistABCD(h):
+    if "mt" in [ax.name for ax in h.axes]:
+        return h[{"passIso" : False, "passMT" : True}]*h[{"passIso" : True, "passMT" : False}].sum().value / h[{"passIso" : False, "passMT" : False}].sum().value
+
+    return sel.fakeHistABCD(h) 
+
+def signalHistSel(h, charge=None, genBin=None):
+    if genBin != None:
+        h = h[{"recoil_gen" : genBin}]
+
+    sel = {"passIso" : True, "passMT": True}
+    if charge in [-1, 1]:
+        sel.update({"charge" : -1j if charge < 0 else 1j})
+    for key in sel.copy().keys():
+        if not key in [ax.name for ax in h.axes]: # remove ax slice if the ax does not exist
+            del sel[key]
+    return h[sel]
+
+class DatagroupsLowPU(Datagroups):
```

Comment:
Same comment, could probably just make an instance of a normal Datagroups class

Reply:
Done